File: benchmarks/oopsla25/subjects/371.py
import sys

def f371(s: str):
  assert len(s) == 1
  if s[0] == "a":
    return
  elif s[0] == "b":
    return
  raise Exception

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  print(f371(my_string))

File: benchmarks/oopsla25/subjects/371.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f371(const char *s) {
    assert(strlen(s) == 1);
    if (s[0] == 'a') {
        return;
    } else if (s[0] == 'b') {
        return;
    }
    fprintf(stderr, "Exception\n");
    exit(1);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f371(my_string);
  return 0;
}
